### PR TITLE
build: update codecov action version

### PR DESCRIPTION
**Pull request by pomegranited**
cf https://discuss.openedx.org/t/codecov-upload-breakages/12877, https://github.com/openedx/axim-engineering/issues/1125

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @pomegranited! Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.

**Comment by kdmccormick** [comment]
closing/opening to see if codecov works now (https://github.com/openedx/axim-engineering/issues/1125)

**Comment by openedx-webhooks** [comment]
@pomegranited Even though your pull request wasn’t merged, please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/edx-submissions&entry.867055334=https://github.com/openedx/edx-submissions/pull/241&entry.1484655318=https://github.com/pomegranited&entry.752974735=2024-05-01+07:43&entry.1917517419=2024-05-14+21:08&entry.2133058324=No) so we can improve your experience in the future.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -38,7 +38,8 @@ jobs:
 
     - name: Run Coverage
       if: matrix.python-version == '3.8' && matrix.toxenv=='django42-drf314'
```
Comment (kdmccormick) [inline .github/workflows/ci.yml:40]:
```suggestion
      if: matrix.python-version == '3.12' && matrix.toxenv=='django42-drf314'
```

**Review by kdmccormick — APPROVED** [review]
Just a suggestion to run Coverage with the latest python, looks good otherwise!

**Comment by pomegranited** [comment]
Since this is just a CI change, the coverage diff failure can be ignored.

**Comment by openedx-webhooks** [comment]
@pomegranited 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/edx-submissions&entry.867055334=https://github.com/openedx/edx-submissions/pull/241&entry.1484655318=https://github.com/pomegranited&entry.752974735=2024-05-01+07:43&entry.1917517419=2024-05-15+23:41&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**